File: 프로그래머스/unrated/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.py
def solution(arr, k):
    num_list = []

    for i, num in enumerate(arr):
        if num not in num_list and len(num_list) < k:
            num_list.append(num)
    
    if len(num_list) < k:
        num_list += [-1]*(k-len(num_list))

    return num_list


arr = [5, 3, 1, 4, 2]
print(solution(arr, 3))

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] arr, int k) {
        List<Integer> result = new ArrayList<Integer>();
        
        // 랜덤으로 주어지는 배열을 반복문 순회
        for(int i=0; i<arr.length; i++) {
            int val = arr[i];
            
            // 배열이 K보다 작은경우
            if(k > result.size()) {
                // 랜덤으로 주어진 값이 포함이 되지 않은 경우
                if(!result.contains(val))
                    result.add(val);   
            }
            
            // 배열이 k와 동일한 사이즈가 된 경우 STOP
            if(result.size() == k)
                break;  
        }
        
        // k사이즈만큼 배열이 채워지지 않았다면
        if(k-result.size() != 0) {
            int cnt = k-result.size();
            // k사이즈 만큼 -1로 채워 넣는다.
            for(int i = 0; i < cnt; i++)
                result.add(-1);
        }
        return result;
    }
}